[Bugfix] PVS Studio issues

TD-er opened the pull request:
Some out of bound bugs may actually have led to crashes or unreliable builds. 
See also #3041

TD-er committed:
[PVS-Studio]  sizeof(Device) -> Device.size()

The 'sizeof (Device)' expression returns the size of the container type, not the number of elements. Consider using the 'size()' function.

TD-er committed:
[PVS-Studio] EventStruct has no operator=

TD-er committed:
[PVS-Studio] Not all members of a class are initialized

TD-er committed:
[PVS-Studio] Missing break; in switch statement (C018 RN2483 TTN)

TD-er committed:
[PVS-Studio] FrasmedOLED display pointer dereference before check !NULL

TD-er committed:
[PVS-Studio] Error log in rules not complete

TD-er committed:
[PVS-Studio] File info struct in webserver rules not all initialized

TD-er committed:
[PVS-Studio] Redundant check in rules conditionMatchExtended

[CWE-571](https://cwe.mitre.org/data/definitions/571.html?utm_source=platformio&utm_medium=piohome):A part of conditional expression is always true: (condOr < condAnd).

TD-er committed:
[PVS-Studio] CWE-1164:The 'note' variable was assigned the same value.

TD-er committed:
[PVS-Studio] ESPEasy p2p incorrect check for binary data

[CWE-571](https://cwe.mitre.org/data/definitions/571.html?utm_source=platformio&utm_medium=piohome):Expression 'packetBuffer[0] != 255' is always true. The value range of char type: [-128, 127].

TD-er committed:
[PVS-Studio] SSDP Specify printf format

[CWE-134](https://cwe.mitre.org/data/definitions/134.html?utm_source=platformio&utm_medium=piohome):It's dangerous to call the 'printf' function in such a manner, as the line being passed could contain format specification. The example of the safe code: printf("%s", str);

TD-er committed:
[PVS Studio] Framed OLED: Simplify check for turning on display

An excessive check can be simplified. The '(A && !B) || (!A && B)' expression is equivalent to the 'bool(A) != bool(B)' expression.

TD-er committed:
[PVS-Studio] Suppress char out of range warning in static data

TD-er committed:
[PVS-Studio] Fix 'dangerous' macro by using braces.

TD-er committed:
[PVS-studio] Fix possible out of range array access

TD-er committed:
[PVS-studio] Suppress char out of range warning in static data

TD-er committed:
[PVS-Studio] Merge duplicate case statements

TD-er committed:
[PVS-Studio] Use implicit cast to float in scheduler idle time stats

TD-er committed:
[PVS-Studio] Remove unused variable in SenseAir plugin

TD-er committed:
[PVS-Studio] Use correct array index in Plugin_Helper_serial.ino

May have been out of bound error on ESP32.

uzi18 commented on src/ESPEasyRules.ino:
```diff
@@ -889,7 +889,7 @@ bool conditionMatchExtended(String& check) {
 
     if ((condAnd > 0) || (condOr > 0)) {                             // we got AND/OR
-      if ((condAnd > 0) && (((condOr < 0) && (condOr < condAnd)) ||
+      if ((condAnd > 0) && (((condOr < 0) /*&& (condOr < condAnd)*/) ||
```
you can just remove this part

TD-er replied:
Yep, but I left it in there as a reference (for you I guess :) )
It is redundant code, as PVS-Studio also noticed.

TD-er commented on src/Networking.ino:
```diff
@@ -135,7 +135,7 @@ void checkUDP()
 
       if (len >= 2) {
-        if (packetBuffer[0] != 255)
+        if (reinterpret_cast<unsigned char&>(packetBuffer[0]) != 255)
```
This one worries me a bit.
Not sure if we have obfuscated similar code where a char is cast into an `uint8_t` so PVS Studio doesn't see it.
On the other hand, maybe we can just trust PVS Studio to have done a thorough check.

TD-er committed:
[ILI9341] Fix build error when building for 1M units

TD-er committed:
[Core 2.7.0] Switch to esp8266/Arduino 2.7.0

TD-er committed:
[Build] Add custom_beta_1M build

Beta:
- Core stage
- `PIO_FRAMEWORK_ARDUINO_ESPRESSIF_SDK22x_191105`

Regular:
- Released core
- `PIO_FRAMEWORK_ARDUINO_ESPRESSIF_SDK22x_190703`

TD-er committed:
[Build] Rename PIO environments referring core_263 to core_270
